### DISCUSSION TITLE
Snap Notifications

### DISCUSSION THREAD

**Discussion by javaspeak** [MetaMask/snaps#2017] [Q&A]
I have some code like this:

```
export const onCronjob: OnCronjobHandler = async ({ request }) => {
  console.log('onCronjob called');
  switch (request.method) {
    case 'checkTransactions':
      return snap.request({
        method: 'snap_notify',
        params: {
          type: 'inApp',
          message: `Hello, world!`,
        },
      });

    default:
      throw new Error('Method not found.');
  }
};
```
Initially I could not see the message "Hello, world!" - then I discovered it had gone to Notifications which is accessible by clicking the 3 dots and then choosing Notifications.  Next to menu item Notifications, it shows you how many unread notifications I have.

However how is the user supposed to know they have a notification without clicking the 3 dots?

Seems strange receiving notifications without being notified?

Is there some setting I need to set in Meta Mask Flask that the user can know if they received a notification without clicking on the 3 dots and choosing Notifications?

Thank you for your time.

**Comment by sai-aarna** [comment]
Hey @javaspeak, I'm currently facing the same issue where the notifcations are inside the notifications settings in metamask and metamask does not pop up. Did you solve this by any means ?

**Reply by javaspeak** [reply]
The only way of letting the user know they have a notification that I could find is to send another notification to the operating system to tell them to look at notifications in their metamask.  

So to do that you send 2 notifications.  One to toApp and the other to native:

type: 'inApp'

type: 'native'

Maybe the snap team will add a little bell to let the user know when they have notifications.

**Reply by sai-aarna** [reply]
ok gotcha, understood this work around but the I think the best can be handled by the snap team itself as the native version has rate limits. 
Thanks 👍

**Comment by Montoya** [comment]
We are aware that this implementation is not perfect. We are working on improving it.

**Reply by javaspeak** [reply]
Thanks :-)
